File: algorithms/Radix Sort/Radix Sort.py
def counting_sort(A, Exp):
    n = len(A)

    count = [0] * (10)

    for i in range(0, n):
        index = A[i] // Exp
        count[index % 10] += 1


    for i in range(1, 10):
        count[i] = count[i]+count[i - 1]

    output = [0] * (n)

    i = n - 1
    while i >= 0:
        index = A[i] // Exp
        output[count[index % 10] - 1] = A[i]
        count[index % 10] -= 1
        i = i- 1

    i = 0
    for i in range(0, len(A)):
        A[i] = output[i]


def radix_sort(A):

    Max = max(A)

    Exp = 1
    while Max / Exp > 0:
        counting_sort(A, Exp)
        Exp *= 10

A = [130, 12, 85, 102, 25, 20, 2]
radix_sort(A)
print(A)

File: algorithms/Radix Sort/Radix Sort.js
function countSort(A, n, Exp) {
    var count = Array.apply(null, Array(10)).map(function () { return 0; });

    for (var i = 0; i < n; i++) {
        count[Math.floor(A[i] / Exp) % 10]++;
    }
    for (var i = 1; i < 10; i++) {
        count[i] = count[i] + count[i - 1];
    }
    var output = Array.apply(null, Array(n)).map(function () { return 0; });

    for (var i = n - 1; i >= 0; i--) {
        output[count[Math.floor(A[i] / Exp) % 10] - 1] = A[i];
        count[Math.floor(A[i] / Exp) % 10]--;
    }

    for (var i = 0; i < n; i++) {
        A[i] = output[i];
    }
}
function RadixSort(A, n) {
    var Max = Math.max.apply(A[0], A);
    var Exp = 1;
    while (Math.floor(Max / Exp) > 0) {
        countSort(A, n, Exp);
        Exp = Exp * 10;
    }
}
var test = [130, 12, 85, 102, 25, 20, 2];
var n = test.length;
RadixSort(test, n);
console.log(test);
